src/schema-validator.ts

```diff
@@ -0,0 +1,139 @@
+import { openmrsFetch } from '@openmrs/esm-framework';
+import { OHRIFormSchema, ConfigObject } from './api/types';
+
+export const handleFormValidation = async (
+  schema: OHRIFormSchema,
+  configObject: ConfigObject,
+): Promise<Array<Array<Record<string, any>>>> => {
+  const errorsArray = [];
+  const warningsArray = [];
+
+  if (schema) {
+    const asyncTasks = [];
+
+    schema.pages?.forEach(page =>
+      page.sections?.forEach(section =>
+        section.questions?.forEach(question => {
+          asyncTasks.push(
+            handleQuestionValidation(question, errorsArray, warningsArray, configObject),
+            handleAnswerValidation(question, errorsArray),
+          );
+          question.type === 'obsGroup' &&
+            question.questions?.forEach(obsGrpQuestion =>
+              asyncTasks.push(
+                handleQuestionValidation(obsGrpQuestion, errorsArray, configObject, warningsArray),
+                handleAnswerValidation(question, errorsArray),
+              ),
+            );
+        }),
+      ),
+    );
+    await Promise.all(asyncTasks);
+
+    return [errorsArray, warningsArray];
+  }
+};
+
+const handleQuestionValidation = async (conceptObject, errorsArray, warningsArray, configObject) => {
+  const conceptRepresentation =
+    'custom:(uuid,display,datatype,answers,conceptMappings:(conceptReferenceTerm:(conceptSource:(name),code)))';
+
+  const searchRef = conceptObject.questionOptions.concept
+    ? conceptObject.questionOptions.concept
+    : conceptObject.questionOptions.conceptMappings?.length
+    ? conceptObject.questionOptions.conceptMappings
+        ?.map(mapping => {
+          return `${mapping.type}:${mapping.value}`;
+        })
+        .join(',')
+    : '';
+
+  if (searchRef) {
+    try {
+      const { data } = await openmrsFetch(`/ws/rest/v1/concept?references=${searchRef}&v=${conceptRepresentation}`);
+      if (data.results.length) {
+        const [resObject] = data.results;
+
+        resObject.datatype.name === 'Boolean' &&
```

Comment:
Should be driven by a config

Reply:
@samuelmale kindly expound on this

Reply:
My bad @arodidev my comment regarding the config, [see](https://github.com/openmrs/openmrs-form-engine-lib/pull/107/files#r1302767224). 

As a good practice, I would create an `enum` holding all the know datatypes to prevent hard-coding these.

```JS

const ConceptDataTypes {
  Boolean: 'Boolean',
  Coded: 'Coded',
  Numeric: 'Numeric',
  Text: 'Text'
  // more...
};


// validator.js


if (concept.datatype.name === ConceptDataTypes.Boolean) {}



```